fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+                    ModularTokenizer.get_special_tokens(self.tokenizers_info[t_type]["json_instance"])
+                )
+                special_tokens_vocab_t = ModularTokenizer.get_subtokenizer_vocab(
+                    tokenizer_json_inst=self.tokenizers_info[t_type]["json_instance"],
+                    token_list=special_tokens_t,
+                )
+
+                if special_tokens_vocab != special_tokens_vocab_t:
+                    result["special token consistency"] = False
+                    result_details["special token consistency"].append(t_type)
+
+            # check if there are no ID collisions within/between vocabs
+            for t_type in tokenizer_types:
+                regular_tokens = list(
+                    ModularTokenizer.get_regular_tokens(self.tokenizers_info[t_type]["json_instance"])
+                )
+                regular_tokens_vocab = ModularTokenizer.get_subtokenizer_vocab(
+                    tokenizer_json_inst=self.tokenizers_info[t_type]["json_instance"],
+                    token_list=regular_tokens,
+                )
+                regular_tokens_IDs = regular_tokens_vocab.values()
+                regular_tokens_ID_set = set(regular_tokens_IDs)
+                if len(regular_tokens_IDs) != len(regular_tokens_ID_set):
+                    result["ID duplicates in vocab"] = False
+                    result_details["ID duplicates in vocab"].append(t_type)
+
+                all_inds_set = all_inds_set.union(regular_tokens_ID_set)
+                if len(all_inds_set) != all_inds_len + len(regular_tokens_ID_set):
+                    result["ID collisions across vocabs"] = False
+                    result_details["ID collisions across vocabs"].append(t_type)
+                all_inds_len = len(all_inds_set)
+
+            special_tokens_ID_set = set(special_tokens_vocab.values())
+            if len(special_tokens_vocab.values()) != len(special_tokens_ID_set):
+                result["ID duplicates in vocab"] = False
+                result_details["ID duplicates in vocab"].append("special")
+
+            all_inds_set = all_inds_set.union(special_tokens_ID_set)
+            if len(all_inds_set) != all_inds_len + len(set(special_tokens_ID_set)):
+                result["ID collisions across vocabs"] = False
+                result_details["ID collisions across vocabs"].append("special")
+            all_inds_len = len(all_inds_set)
+
+        return result, result_details
+
+    def is_consistent(self) -> bool:
+        """Returns True if the modular tokenizer is consistent, i.e.:
+        a. Special tokens are the same (and map to the same indices) across all the tokenizers
+        b. Regular token ID mappings of any given tokenizer do not collide with special token mappings, nor with ID mappings of other tokenizers
+
+        Raises:
+            Exception: _description_
+
+        Returns:
+            bool: _description_
+        """
+        raise Exception("Not implemented")
```

Comment:
Are you planing to implement it? if not, open a trello card.

Reply:
Implemented.